content/Software and Downloads/Upload/Error-avrdude-when-uploading.md

```diff
@@ -63,17 +92,83 @@ avrdude: stk500_getsync() attempt 2 of 10: not in sync: resp=0x00
+> If you're using a classic Arduino Nano you also need to [select the correct processor](https://support.arduino.cc/hc/en-us/articles/4401874304274-Select-the-right-processor-for-Arduino-Nano).
 
 <!-- TODO: Update when dedicated article is deployed -->
 
 If the issue persists, [check these steps](#check-the-following).
 
-<h3 id="avrdude-butterfly_recv">avrdude: butterfly_recv()</h2>
+---
+
+<a name="stk500v2_ReceiveMessage_getsync"></a>
+
+## avrdude: stk500v2_ReceiveMessage(): timeout<br>avrdude: stk500v2_getsync(): timeout communicating with programmer
+
+<!-- NOTE: Occurs when uploading to Arduino Uno with "Arduino ATmega2560 (Mega 2560)" as selected board. -->
+<!-- NOTE: Occurs when uploading to Arduino Nano with "Arduino ATmega2560 (Mega 2560)" as selected board. -->
+
+Make sure you have [selected the right board and port](https://support.arduino.cc/hc/en-us/articles/4406856349970-Select-board-and-port-in-Arduino-IDE) correct board under _Tools > Board_ and  _Tools > Port_.
+
+---
+
+<a name="avrdude-ser_open-cant-open-device-no-such-file-or-directory"></a>
+
+## avrdude: ser_open(): can't open device: No such file or directory
+
+```
+avrdude: ser_open(): can't open device "/dev/cu.usbmodem14101": No such file or directory
+Problem uploading to board. See http://www.arduino.cc/en/Guide/Troubleshooting#upload for suggestions.
+```
+
+<!-- NOTE: This error can occur when the selected port has become unavailable, e.g. board being disconnected after the port was selected. -->
+<!-- NOTE: Can probably be caused by the issue with COM ports being reassigned in Windows -->
+
+1. Make sure the correct port is selected in _Tools > Port_.
+
+   * For step-by-step instructions, see [Select board and port in Arduino IDE](https://support.arduino.cc/hc/en-us/articles/4406856349970-Select-board-and-port-in-Arduino-IDE).
+   * If you can't find the port, see [If your board does not appear in the port menu](https://support.arduino.cc/hc/en-us/articles/4412955149586-If-your-board-does-not-appear-in-the-port-menu).
+
+2. If using the classic Nano, try selecting a different processor in _Tools > Processor_.
+
+---
+
+<a name="ser_open-resource-busy"></a>
+
+## avrdude: ser_open(): can't open device <port>: Resource busy
+
+<!-- NOTE: Occurs when uploading to /dev/cu.wlan-debug with Arduino Every selected -->
+
+1. Make sure the correct port is selected in _Tools > Port_.
+
+2. Close other instances of Arduino IDE, serial monitors or other software that my be blocking the port.
+
+3. Restart your computer.
+
+4. [Find and stop process blocking a port](https://support.arduino.cc/hc/en-us/articles/4407830972050-Find-and-stop-process-blocking-a-port).
+
+---
+<a name="ser_open-permission-denied"></a>
+
+## avrdude: ser_open(): can't open device <port>: Permission denied
+
+<!-- * -->If you get this error on Linux, [make sure your user account is part of the dialout group](https://support.arduino.cc/hc/en-us/articles/360016495679).
```

Comment:
This is not showing up as a link in rich diff view.

Reply:
50973d56e1c6fb3498fd5c7952b66e3163155359